src/integer/poly_over_z/from.rs

```diff
@@ -83,6 +86,40 @@ impl FromStr for PolyOverZ {
     }
 }
 
+impl From<&PolyOverZq> for PolyOverZ {
+    /// Create a [`PolyOverZ`] from a [`PolyOverZq`].
+    ///
+    /// Parameters:
+    /// - `poly`: the polynomial from which the coefficients are copied
+    ///
+    /// Returns representative polynomial of the [`PolyOverZq`] element.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::PolyOverZ;
+    /// use qfall_math::integer_mod_q::PolyOverZq;
+    /// use std::str::FromStr;
+    ///
+    /// let poly = PolyOverZq::from_str("4  0 1 102 3 mod 101").unwrap();
+    ///
+    /// let poly_z = PolyOverZ::from(&poly);
+    ///
+    /// # let cmp_poly = PolyOverZ::from_str("4  0 1 1 3").unwrap();
+    /// # assert_eq!(cmp_poly, poly_z);
+    /// ```
+    fn from(poly: &PolyOverZq) -> Self {
+        let mut out = Self::default();
+        unsafe {
+            fmpz_mod_poly_get_fmpz_poly(
+                &mut out.poly,
+                &poly.poly,
+                poly.modulus.get_fmpz_mod_ctx_struct(),
+            )
```

Comment:
This returns the coefficience with the modulus.
a) is this something we should do?
e.g. `"2  2 12 mod 10"` => `"2  2 12"` or `"2  2 2"`
b) should we do it here or when we create the polynomial?

Regardless, the behavior has to be documented more explicit. Maybe also simplify the doc test example and make the test assert visible.

Reply:
I am not quite shure what you mean. We return the representative polynomial under having applied the modulo operation for each coefficient. If we now create a PolyOverZ from a PolyOverQ, I would expect exactly this behaviour, i.e. to get the polynomial representing the PolyOverZq.
I simplified the doc example, but I think it was already quite clear before

Reply:
I think that the added "(all reduced coefficients)" and simplified comment helps to understand it better. :+1: